### ISSUE TITLE
Thank you so much

### ISSUE THREAD

**Issue by ryancwalsh** [antelle/generate-ip-cert#1]
Thank you so much for providing your helpful script, which I found at https://medium.com/@antelle/how-to-generate-a-self-signed-ssl-certificate-for-an-ip-address-f0dd8dddf754

I've been trying to figure out how to modify it to be usable on PC, Mac, iOS, and Android.

Previously, I'd been using https://stackoverflow.com/a/57684211/470749

But that seems unable to work with an IP address.

Do you happen to know how I could modify that "One self-signed cert to rule them all" to use an IP address rather than hostname?

In other words, what I seem to need is a merge of your approach and ScottyB's.

I appreciate your thoughts. Thanks.

**Comment by antelle** [comment]
Hi! You mean, you expect the script to be able to run on all those OS, or you noticed an issue with certificates generated by it?

**Comment by ryancwalsh** [comment]
What a fast response! Thanks!

I mean that I was not able to install the certificate at Android > Settings > Security > Advanced > Encryption & credentials > Install a certificate > CA Certificate.

It says "Private key required to install a certificate".

The only way I've ever been able to get a self-signed cert to be installable on Android is https://stackoverflow.com/a/57684211/470749

I don't understand all the particulars about why each type of device (Android, iOS, PC, Mac) requires different lines in the conf file or whether it would be possible to use an IP instead of hostname.

Thanks.

**Comment by antelle** [comment]
Interesting, I won't be able to check if it works on Android, but I can see what happens on iOS.
